test/compiler/interpreter_exec.jl

```diff
@@ -106,3 +106,19 @@ let m = Meta.@lower 1 + 1
     global test29262 = false
     @test :b === @eval $m
 end
+
+@testset "many basic blocks" begin
+    n = 1000
+    ex = :(return 1)
+    for _ in 1:n
+        ex = :(if rand()<.1
+            $(ex) end)
+    end
+    @eval begin
+        function f_1000()
+            $ex
+            return 0
+         end
+    end
+    @test f_1000()===0
+end
```

Comment:
I'm confused at this test case. What's being tested here?

Reply:
I believe that this is testing compilation of a function with a bunch of basic blocks. Is there a better version of this test?

Reply:
Is the compilation performance being tested, or is there any correctness issue? And why `f_1000` always returns `0`?

Reply:
There was a correctness issue (https://github.com/JuliaLang/julia/issues/47065) where the compiler would try to sort a big list and fail. `f_1000` doesn't technically always return 0, but it returns 1 1 time in 10^1000 which is close enough to 0 that it will never fail.

Reply:
Okay, I understand. I'd prefer not having `@testset` there though.